### PR TITLE
Add service discoverability for backend routes

### LINKED ISSUE #2973: [bug] signing into frontend is not possible after cloud.gov maintenance.
Thank you for taking the time to let us know about the issue you found. The basic rule for bug reporting is that 
something isn't working the way one would expect it to work. Please provide us with the information requested 
below and we will look at it as soon as we are able.

## Description

As noted in #2839, cloud.gov's maintenance schedule mangles the routes to TDP apps. This includes the route between the frontend and the backend. After maintenance occurs, when users attempt to sign into TDP, they are redirected to the following screen:

maintlogin
 Admins have to restart the frontend to resolve this and are not aware when this happens unless continuously monitoring access to the system. 

## Action Taken

- navigating to the frontend website via browser
- clicking either of the sign-in buttons
loginscreen2

## What I expected to see

I expected to see one of the two sign-in screens:
loginscreen

## What I did see

maintlogin

## Other Helpful Information

+ URL of the page I was on:
+ Browser and version: chrome, edge, firefox, etc. 
+ Operating System: ( Windows | MacOS X | Linux | Other )
+ Is the issue repeatable?: ( yes | no | don't know )
+ Has the issue occurred more than once?:

### PR DESCRIPTION
**Pull request by jtimpe**
## Summary of Changes
Pull request closes #2973 
* Uses the same approach as #2903 did for kibana/clamav, applied to the backend and `kibana_auth_check` routes.

## How to Test
1. Deploy to a cloud.gov environment
1. Load the frontend and click the log in link. You should navigate to login.gov login
1. Restart the backend app in cloud.gov.
1. Reload the frontend. Click the log in link. You should still navigate to login.gov login

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [ ] fix the service discoverability bug
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by andrew-jameson — APPROVED** [review]

**Review by elipe17 — APPROVED** [review]
Worked for me locally. Has this been tested in a deployed env?

**Comment by jtimpe** [comment]
> Worked for me locally. Has this been tested in a deployed env?

@elipe17 yes, but i would appreciate a second look. happy to deploy it to raft again for you if you'd like

File: tdrs-frontend/nginx/local/locations.conf [hunk 1]
```diff
@@ -4,9 +4,10 @@ location = /nginx_status {
     deny all;
 }
 
-location ~ ^/(v1|admin|static/admin|static/drf-yasg|swagger|redocs) {
+location ~* ^/(v1|admin|static/admin|static/drf-yasg|swagger|redocs)/(.*)$ {
```
Comment (raftmsohani) [inline tdrs-frontend/nginx/local/locations.conf]: why did we have to do this change exactly? the existing regex should capture all locations with trailing URI sections

Comment (jtimpe) [reply tdrs-frontend/nginx/local/locations.conf]: good call. i saw this change in the original pr and thought it was needed, but i tested locally and seems to work without it. removed.

**Review by raftmsohani — APPROVED** [review]
LGTM!

**Review by ADPennington — APPROVED** [review]
lgtm @jtimpe 🚀 

- I tested sign-in via both authentication services and after both a restart and restage of the backend.

**Pull request merged**